pkg/doc/signature/proof/jws.go

```diff
@@ -0,0 +1,183 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package proof
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"strings"
+
+	"github.com/piprate/json-gold/ld"
+)
+
+const securityContext = "https://w3id.org/security/v2"
+
+// CreateDetachedJWTHeader creates detached JWT header.
+func CreateDetachedJWTHeader(p *Proof) string {
+	jwtHeaderMap := map[string]interface{}{
+		"alg":  p.Type,
+		"b64":  false,
+		"crit": []string{"b64"},
+	}
+
+	jwtHeaderBytes, _ := json.Marshal(jwtHeaderMap) //nolint:errcheck
+
+	return base64.RawURLEncoding.EncodeToString(jwtHeaderBytes)
+}
+
+// GetJWTSignature returns signature part of JWT
+func GetJWTSignature(jwt string) ([]byte, error) {
+	jwtParts := strings.Split(jwt, ".")
+	if len(jwtParts) != 3 || jwtParts[2] == "" {
+		return nil, errors.New("invalid JWT")
+	}
+
+	return base64.RawURLEncoding.DecodeString(jwtParts[2])
+}
+
+func getJWTHeader(jwt string) (string, error) {
+	jwtParts := strings.Split(jwt, ".")
+	if len(jwtParts) != 3 { // nolint:gomnd
+		return "", errors.New("invalid JWT")
+	}
+
+	return jwtParts[0], nil
+}
+
+// createVerifyJWS creates a data to be used to create/verify a digital signature.
+// By the time of development, no specification was found with the algorithm description.
+// Current implementation is based on the https://github.com/digitalbazaar/jsonld-signatures.
```

Comment:
Did you have a chance to look at this spec: https://tools.ietf.org/html/rfc7797?

Reply:
Yes. I will add a reference to this spec. At https://github.com/digitalbazaar/jsonld-signatures, they use several heuristics which I did not found in any spec:
1) Use https://w3id.org/security/v2 as a `@context` for JSON-LD canonization of both JSON and Signature documents.
2) Apply preliminary JSON-LD compacting of JSON document using the same https://w3id.org/security/v2 `@context`.
3) Remove only `jws` and `proofValue` elements only from the proof document before canonization. At `signature` package (and what is stated in https://w3c-dvcg.github.io/ld-signatures/#create-verify-hash-algorithm), we remove `id`, `type`, `proofValue` and `jws` elements.